server/src/error.rs

```diff
@@ -20,20 +56,54 @@ impl IntoResponse for CheeseError {
     type BodyError = <Self::Body as axum::body::HttpBody>::Error;
 
     fn into_response(self) -> axum::http::Response<Self::Body> {
-        let bb = match self {
-            Self::Mongo(e) => {
-                event!(Level::ERROR, "MongoDB error: {:?}", e);
-                Self::Body::from("MongoDB did a fuckywucky")
-            }
-            Self::Oid(e) => {
-                event!(Level::ERROR, "ObjectID error: {:?}", e);
-                Self::Body::from("ObjectID did a fuckywucky")
-            }
-        };
-
-        Response::builder()
-            .status(StatusCode::INTERNAL_SERVER_ERROR)
-            .body(bb)
-            .unwrap()
+        match self {
+            Self::Mongo(e) => response_error!(e, internal server error),
+            Self::Oid(e) => response_error!(e, internal server error),
+            CheeseError::Login(l) => match l {
+                e @ LoginError::InvalidCredentials => {
```

Comment:
Is there a specific reason for using this `e @ ... ` syntax over just using a `&a` ?
This feels, at face value, like un-needed complexity wrt how easy the code base is to read.

Reply:
I want to match the entire error to give the right error message, but also want to print the original error message.